src/shared/lib/url-sanitization.ts

```
@@ -0,0 +1,282 @@
+  // If no protocol specified, default to https
+  let urlToValidate = trimmedUrl;
+  if (!trimmedUrl.includes('://')) {
+    urlToValidate = `https://${trimmedUrl}`;
+  }
+
+  try {
+    // Parse URL to validate structure
+    const parsedUrl = new URL(urlToValidate);
+    
+    // Validate protocol
+    if (!ALLOWED_PROTOCOLS.includes(parsedUrl.protocol as any)) {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)